crates/crypto/src/hashers.rs

```diff
@@ -0,0 +1,75 @@
+use {
+    crate::truncate,
+    blake2::{Blake2b512, Blake2s256},
+    digest::Digest,
+    sha2::{Sha256, Sha512},
+    sha3::{Keccak256, Sha3_256, Sha3_512},
+};
+
+pub fn sha2_256(data: &[u8]) -> [u8; 32] {
+    Sha256::digest(data).into()
+}
+
+pub fn sha2_512(data: &[u8]) -> [u8; 64] {
+    Sha512::digest(data).into()
+}
+
+pub fn sha2_512_truncated(data: &[u8]) -> [u8; 32] {
+    truncate(&sha2_512(data)).unwrap()
+}
+
+pub fn keccak_256(data: &[u8]) -> [u8; 32] {
+    Keccak256::digest(data).into()
+}
+
+pub fn sha3_256(data: &[u8]) -> [u8; 32] {
+    Sha3_256::digest(data).into()
+}
+
+pub fn sha3_512(data: &[u8]) -> [u8; 64] {
+    Sha3_512::digest(data).into()
+}
+
+pub fn sha3_512_truncated(data: &[u8]) -> [u8; 32] {
+    truncate(&sha3_512(data)).unwrap()
+}
+
+pub fn blake2s_256(message: &[u8]) -> [u8; 32] {
+    Blake2s256::digest(message).into()
+}
+
+pub fn blake2b_512(message: &[u8]) -> [u8; 64] {
+    Blake2b512::digest(message).into()
+}
+
+pub fn blake3(message: &[u8]) -> [u8; 32] {
+    blake3::hash(message).into()
+}
+
+// ----------------------------------- tests -----------------------------------
+
+#[cfg(test)]
+mod tests {
+    use hex_literal::hex;
+
+    macro_rules! hash_test {
+        ($hash_fn:ident, $world:expr => $hexed:expr) => {
+            #[test]
+            fn $hash_fn() {
+                let hash = super::$hash_fn($world.as_bytes());
+                assert_eq!(hash, hex!($hexed))
```

Comment:
Use `hex_literal::hex!` which decodes the HEX strings at compile time. Also avoid casting `hash` to a `Vec`.